opm/autodiff/FullyImplicitBlackoilSolver_impl.hpp

```diff
@@ -1254,6 +1275,17 @@ namespace {
         struct Chop01 {
             double operator()(double x) const { return std::max(std::min(x, 1.0), 0.0); }
         };
+
+        double infinityNorm( const ADB& a )
+        {
+            if( a.value().size() > 0 ) {
+                return a.value().matrix().template lpNorm<Eigen::Infinity> ();
```

Comment:
> Actually it was correct in the initial version since there it was in a template context

That's a necessary condition, not sufficient.  The `.template` construct is only appropriate if the expression of which it is part depends on a template parameter.  That was not the case in the original version either.

> I am surprised that 4.8 does not complain, my clang accepts it also. Does that indicate that this has been relaxed a little in C++11 or C++14?

Possibly.  I haven't read up on all the details.